### PR TITLE
add response and uri field to RetryableStatusCodeException

### PR DESCRIPTION
**Pull request by lly5044**
#271 
this PR help to fix the problem of spring-cloud-netflix #2353

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@lly5044 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=272)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=272) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@lly5044 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=272)!

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryableStatusCodeException.java [hunk 1]
```diff
@@ -10,7 +11,25 @@
 
 	private static final String MESSAGE = "Service %s returned a status code of %d";
 
+	private Object response;
+
+	private URI uri;
+
 	public RetryableStatusCodeException(String serviceId, int statusCode) {
```
Comment (ryanjbaxter) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryableStatusCodeException.java]: Probably want to mark this as deprecated

**Comment by lly5044** [comment]
I fix the RetryLoadBalancerInterceptor
@ryanjbaxter

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java [hunk 2]
```diff
@@ -69,27 +77,97 @@ public ClientHttpResponse intercept(final HttpRequest request, final byte[] body
...
+                            RetryableStatusCodeException ex = (RetryableStatusCodeException) lastThrowable;
+                            return (ClientHttpResponse) ex.getResponse();
+                        }
+                        if (lastThrowable instanceof Error) {
+                            throw (Error) lastThrowable;
+                        }
+                        if (lastThrowable instanceof Exception) {
+                            throw (Exception) lastThrowable;
+                        } else {
+                            throw new RetryException("Exception in retry", lastThrowable);
```
Comment (ryanjbaxter) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java]: Do we need to wrap it?

**Comment by ryanjbaxter** [comment]
@lly5044 the master branch is now version 2.0.x for Finchley.  If you would like this to go in the next Edgware release you will need to submit the PR against the 1.3.x branch

File: spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java (hunk 2 above)
Comment (lly5044) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java]: I get a Throwable from retryContext but I can only throw an Exception here, so I wrap it with RetryException.

Comment (ryanjbaxter) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java]: Maybe it is best to wrap the `Throwable` in a `RetryException` regardless and use the message, `Could not recover`.

Comment (lly5044) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/RetryLoadBalancerInterceptor.java]: Maybe you are right

**Pull request merged**